datastructs/ring.h

```diff
@@ -9,20 +19,87 @@
 
 struct ring{
         /* Fill this in */
+        int size;
+        int pop;
+	int push;
+        ptr_t *items;
+}
+
+int isEmpty(struct ring *rb)
```

Comment:
check the [style guide's requirements for naming](https://github.com/gwAdvNet2015/adv-net-samples/blob/master/styleguide.md#variable-and-function-naming)--you shouldn't use this kind of "camelCase" which is common in java. We want all function names in the library to be similar and follow an easy to use pattern.  Change this to something like `ring_is_empty()`.